### PR TITLE
refactor(FOROME-1185): unify modal and panel for compound het

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-1185

### CODE REVIEW

File: src/components/conditions/compound-het-condition/compound-het-condition.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          approx: getApproxValue(approx),
+        }),
+      })
+
+      return () => statFuncStore.reset()
+    }, [approx, statFuncStore])
+
+    return (
+      <>
+        <>
```
Comment (AlMaQntr) [inline src/components/conditions/compound-het-condition/compound-het-condition.tsx]: Double <>?

File: src/components/conditions/inheritance-mode-condition/components/inheritance-mode-variants.tsx [hunk 2]
```diff
@@ -37,7 +38,7 @@ export const InheritanceModeVariants = ({
         ))
       ) : (
         <div className="flex justify-center items-center text-grey-blue my-2">
-          {t('dtree.noFilters')}
+          {status || t('dtree.noFilters')}
```
Comment (AlMaQntr) [inline src/components/conditions/inheritance-mode-condition/components/inheritance-mode-variants.tsx]: inheritanceMode namespace for i18n?

File: src/components/conditions/inheritance-mode-condition/inheritance-mode.interface.ts [hunk 3]
```diff
@@ -1,48 +1,45 @@
...
+  handleSetProblemGroups: (checked: boolean, problemGroup: string) => void
 }
 
 export interface IInheritanceModeVariantsControlsProps {
   selectedVariants: string[]
-  selectAllVariants: () => void
-  clearAllVariants: () => void
   attributeSubKind: string | undefined
   mode: ModeTypes | undefined
+  selectAllVariants: () => void
```
Comment (AlMaQntr) [inline src/components/conditions/inheritance-mode-condition/inheritance-mode.interface.ts]: on... prefix for callbacks

**Review by AlMaQntr — APPROVED** [review]

File: src/pages/filter/dtree/components/modals/components/modal-compound-het/modal-compound-het.tsx [hunk 4]
```diff
@@ -1,100 +1,90 @@
...
-          currentGroup={currentGroupToModify ?? currentStepGroups}
-          handleAddAttribute={handleAddAttribute}
-        />
-      )}
+      <CompoundHetCondition
+        initialApprox={initialApprox}
+        initialMode={initialMode}
+        attributeSubKind={attributeSubKind}
+        statFuncStore={dtreeStatFuncStore}
+        controls={({ hasErrors, param, mode }) => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/dtree/components/modals/components/modal-compound-het/modal-compound-het.tsx]: move to a separate function

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**